src/validate.ts

```diff
@@ -0,0 +1,63 @@
+import { IMCFormModel } from "./components/Formulario";
+import { ErrorObject } from "./model";
+
+const required = "Campo obrigatório.";
+const onlyNumbers = "Deve ser um número.";
+const greaterThanZero = "Deve ser um valor maior do que 0.";
+
+export function validate(errors: ErrorObject<IMCFormModel>, values: IMCFormModel) {
+  if (!values.altura) {
+    errors.altura = required;
+  } else {
+    errors = validateAlturaOnlyNumbers(errors, values);
+    errors = validateAlturaGreaterThanZero(errors, values);
+  }
+
+  if (!values.peso) {
+    errors.peso = required;
+  } else {
+    errors = validatePesoOnlyNumbers(errors, values);
+    errors = validatePesoGreaterThanZero(errors, values);
+  }
+  return errors;
```

Comment:
Alguns pontos que acho interessante ressaltar:
- Se você observar os métodos `validate...OnlyNumbers` e `validate...GreaterThanZero` eles seguem a mesma estrutura, a única coisa que muda é qual **atributo** (**_chave de um objeto/tipo_**) do `values` é verificado e para qual **atributo** do `errors` é atribuída a mensagem de erro --> Isso poderia virar algo mais **genérico** para você conseguir usar em qualquer campo de qualquer formulário e reduziria drasticamente a quantidade de funções necessárias para validar também 
- Acho que seria melhor criar uma função `validateRequired` porque o comportamento é muito parecido e assim você evitaria uma cópia de código em todos os validadores que necessitarem verificar isso, bastaria usar a função

Existem dois caminhos para você tratar a questão de depender da required para não validar as outras regras:
1. As próprias funções de validação podem fazer isso (se o valor for `null` ou `undefined`, basta não validar e seguir em frente)
2. Adicionar uma flag, tipo `isValid`, no seu `ErrorObject` para controlar se está válido ou não, aí bastaria dar um `if (errors.isValid) para chamar os outros validadores (que nem no back)

A opção 1 é mais fácil de implementar, enquanto a 2 teria vantagens na hora de testar (bastaria dar um assert true no `errors.isValid`)

Reply:
Sobre o `validate...GreaterThanZero`, acredito que esse método poderia ser um pouco mais genérico para aceitar qualquer valor e adaptar a mensagem de erro conforme o parâmetro recebido
O que você acha ?

Reply:
Sobre os pontos que comentei ali em cima, você vai precisar entender os genéricos no Typescript
Seguem algumas referências:
1. [Documentação sobre genéricos](https://www.typescriptlang.org/docs/handbook/2/generics.html)
2. [Playground para brincar com os genéricos](https://www.typescriptlang.org/pt/play?#code/PTAEBUEsAcHsGdQHMCmA7AlwJ0gYwaNClgLaQAuKJoArvAIZYQwH4mygBujkAh5ykiIqoWDXJYC5FvAB0AWABQIUAFkU5ekqUrw6TVioES9NJSYp4lUCgAeVaABsOjlANmgAarFwArwrAAJiigAGb0AF7EOmAkNBT0oPhCiWyg0nCIqJg4+MKgGISSnPwojqAAjjQhAgBWifSOSDTUuADPgZBIHABGVDFJsCROePSBHHSm4+kyyOjYeAQijc3UcGYcALSgJohC8LADaADHHPBdaJCho5XVM5lzOYuI8KeDw46j43LaymCe9HYiDYKAA54x3rRqHYHM5RGhgjY0JxYI4BIYgVDRD1ahoOIj9Fgxlo-jZqC1QIwiQBPDwAZTaXEBBDQBEgw1gWAMkJaiW4OFK5TQlmsuHoBwAXAMOBkOFUQqFYJBCOKGNM2BxGKt9LApYolKEaGhcNJYGgkSi0cQAKIkACCWBpAB4oHAAHwACkJxIlzDgAEpfa7YABtAC6oAA3kpQKBDOQaFhzSHvYF6GGANxKAC+vxUADlYJpfYU2BT0X6zsQ4yhQsRIOqhhxwB4AJLwA4FAa4GiOBNExwkfQhc5kRwQ1JNymIIgd2Di2gMajKkQU5ywaBhTkDaCMRKGaCGeD6CEAW4ReBQHnAoFZpEaQ7MIQPR5PldAHvA1KI-oANEjyFOAZJgxAgOFZId3icPEVSJHZ6EgTkknHQwIhJAZCw7RJDTQUE3hA3hQEgNA6ywXxjUgRJdzg49hkMeCnyYGUWB3VUxg4DVKSwbUNiI5pGkpZi4BrchOTQdiPzgLAMBIAZ0EtTgGw4VdqCpehqX9X58DQKxQCsHA0CQB0aVAABeC1UXRW1jPUj0ACIAAsyjhbJ61weA7P9LNFG03S0BaPosBs6kzIsq0sGsx1bIARgAJgAZi8vMwAABSCIxEHRK5RiYeUiLnMIjVwRDxPeZT4CIIlpnoDinLFLABmPAJgkxegJEgHp4iYCk1JCxF9OIpBEESXlCtwt55QGYIssSY9utUqK+pCWAcTxeAlF88hbxq4KABEUDpdrDPgX0BsM8NQvkyybXtRaPUjbMkv1UlvD8dJAR6GTmpCNwEiYRJiNI8jioaTpirNKZDlJISOBRN6h3gdg9U2riaX2w6DKG2LLuRa6Itu50zqQT07M87y5Nx8LIpMyqyGsPKqkaeUmFlPSUHqRd2P-KSvrsIRKFk0kxQORAzUROH-EPFBioYQhiDpyB-qIcTEApeAaB6bTaiNUTQERWUflJfNLGPYFxQICXvr1yAolyu4Yb1twfqsdBgnmgZWcsIhcFPa58AUZ7iPMcJcBCfbjzQBz6AEcpo0UWNggjqOsF9D1-TMt0uCVQJvNzZ6VGtKxsKNPCOEMUO+ixLWdfxZbVtExA8sydWVOL8aoZUajEkT9Bk+29IyhJHCTRKmsEXrNCsHzeg9HHF0WBsWxKARRBw976Oyk9VN6CDFhw3TuPY232RFQi+hcAcj1lVMzPIFkHvI8YNOnrzjazV0lbcQbgBhIY18fpg5kQyRkdknRgqd043yjNmUA2Z-wgIfsnCBGdoGwMzEoQw48cCT2nrPegHpP5rV-iQf+ycnoDHbJ2cIjhk6ESaogiEJxyrSArFLCOlA9QYNdhPRguDB4emAXAqMoD14pw-JAzOD00HkNJMGLI8xcgEDgC1d4YIJywWltWNgIwxTfFuJDLOb0BYDBIAAL77DAccmUeClCEB4O0VQVxWEIhSGEHxa5QjbqXR2SRz4OUGgMJu8sKAhDypbZeUdrakWHMCM02sNiIH1rMAkZgqokhUCEMUF9LAByUEHYgIcQgAAkEDkH2t-PxIRD56UaPyVOO9KRoA0iglEDZc6-GHqac0Yw8AlUYAAeXrrAO0sAKnZPnnAf8YynKL2XoERAJSrDlMqZ6T+u9Jm+Oyb6aZKBAygB2VGGMmynKyAYGiJ+n8nqxnjImc0WSnLtPzmAQu1hCgcARuwResI5R3HPuyeg3NORfTSBSaiAAjocEhlJLyJAHAsRYGldzCJEXwmw4TDz6VgQF0kSD-hYRlW4IQQKNNpKAShHcwDKJHNWI8lklKcwRBMEwRFGKFMeAsMUvx8lYDZYsspKAdkAHF0BuVgBM2A5SzSUBoOMSRRyzl1I-A04MkqnwytgBIrObSczJVAHabocFGaMtEKAPlyzsnCuFIogonNxj-kttNEInRbYDG7mUUJNABJvMrKq6V4wOlFS6ZSMGmLBlf1gK6diIydniqmZU2ZrsFmlPNU5S1orxVulWTidZsB-z3JQNsypeyDnVPzac2pFycRXJErc45KBHkDH6V89xlJirMsKH0ZcT4dLKWoOYTE-Ug70HsB4NKOLfwDH2BwQ8sBOD+FGseZoRJcKJBMIhLAHhC4MBeG8Sw+AaELkmqSegBq+Q+H8ALQYkdpbsh1LBbCkRiAeAAIqeuNafIos7qx2hSu2V1ohxBEhnCbec8A8WzGyBysC0TQ7kmoLiCgHBQiuFsKeGOPjaYUEVoEu4g9Va0TZnQIobRDyIUQAbNmfbqx5UBvWJgx4aBJA6F0adSFtGuFErk0kKU9zwSEM2pwBAkkPEg4o2JrQrGmxsCy4O59LB6hAHJWwgJoISjGNwY0KBAibFwJJywSmVOuAlOQb8e6cDQHIJsAA7hQBymxDDn3IAMBy5ByDQBOiAKzXnZAmdnLgcz5BxyGVkJyJAwBxjuWAFHBEPRYCwAANbAFcrkOQLmSCOCUEAA)
3. [Documentação sobre o operador keyof](https://www.typescriptlang.org/docs/handbook/2/keyof-types.html)
4. [Tutorial de como usar as keys de um Object para pegar os valores dos atributos](https://bobbyhadz.com/blog/typescript-object-get-key-by-value)